File: bench/template/bg/nsieve/nsieve.py
import json
import time


def nsieve(n):
    xs = [0] * (n + 1)
    count = 0
    i = 2
    while i <= n:
        if xs[i] == 0:
            count += 1
            j = i * i
            while j <= n:
                xs[j] = 1
                j += i
        i += 1
    return count


n = {{ .N }}
repeat = 50
last = 0

start = time.perf_counter()
for _ in range(repeat):
    last = nsieve(n)
duration = (time.perf_counter() - start) * 1e6

print(json.dumps({
    "duration_us": duration,
    "output": last,
}))

File: bench/template/bg/nsieve/nsieve.lua
local function nsieve(n)
  local xs = {}
  for i = 0, n do xs[i] = 0 end
  local count = 0
  local i = 2
  while i <= n do
    if xs[i] == 0 then
      count = count + 1
      local j = i * i
      while j <= n do
        xs[j] = 1
        j = j + i
      end
    end
    i = i + 1
  end
  return count
end

local n = {{ .N }}
local repeats = 50
local last = 0

local start = os.clock()
for _ = 1, repeats do
  last = nsieve(n)
end
local duration_us = (os.clock() - start) * 1e6

io.write(string.format('{"duration_us": %d, "output": %s}\n', math.floor(duration_us), tostring(last)))
